examples/kmeans/kmeans.py

```diff
@@ -0,0 +1,88 @@
+import six
+try:
+    import matplotlib
+    matplotlib.use('Agg')
+    import matplotlib.pyplot as plt
+except ImportError:
+    pass
+
+import numpy as np
+
+import cupy
+
+
+def fit(X, n_clusters, max_iter, elem):
+    assert X.ndim == 2
+    xp = cupy.get_array_module(X)
+    pred = xp.zeros(len(X), dtype=np.int32)
+    initial_indexes = np.random.choice(len(X), n_clusters,
+                                       replace=False).astype(np.int32)
+    centers = X[initial_indexes]
+    data_num = X.shape[0]
+    data_dim = X.shape[1]
+
+    for _ in six.moves.range(max_iter):
+        # calculate distances and label
+        distances = xp.zeros((data_num, n_clusters), dtype=np.float32)
+        if elem is False or xp == np:
+            distances = xp.linalg.norm(X[:, None, :] - centers[None, :, :],
+                                       axis=2)
+        else:
+            cupy.ElementwiseKernel(
```

Comment:
When you use ElementwiseKernel, memory you need gets smaller probably. However, it may be better to delete them because the current ElementwiseKernel code looks complicated and we can use it only when `n_clusters == 2`